### ISSUE TITLE
v7.3.1 -  amplify pull failed

### ISSUE THREAD

**Issue by kouz75** [aws-amplify/amplify-cli#8871]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.x

### Amplify CLI Version

7.3.1

### What operating system are you using?

macos monterey

### Amplify Categories

Not applicable

### Amplify Commands

pull

### Describe the bug

amplify crash when pulling a project.
```
Failed to pull the backend.
Found '{' where a key name was expected (check your syntax or use quotes if the key name includes {}[],: or whitespace) at line 67,5 >>>    {
      "rootSta ...
Error: Found '{' where a key name was expected (check your syntax or use quotes if the key name includes {}[],: or whitespace) at line 67,5 >>>    {
      "rootSta ...

```

### Expected behavior

the project was working fine with previous cli v6.3 

### Reproduction steps

just pull a project:
amplify pull --appId xxx --envName dev

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
amplify pull --appId xxxx --envName dev
✔ Successfully received Amplify Admin tokens.
Amplify AppID found: xxxx. Amplify App name is: xxxx
Backend environment dev found in Amplify Console app: xxxx
? Choose your default editor: None
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using none
? Source Directory Path:  src
? Distribution Directory Path: dist
? Build Command:  npm run-script build
? Start Command: npm run-script start
? Do you plan on modifying this backend? Yes
✔ Successfully pulled backend environment dev from the cloud.
✖ There was an error initializing your environment.
Failed to pull the backend.
Found '{' where a key name was expected (check your syntax or use quotes if the key name includes {}[],: or whitespace) at line 67,5 >>>    {
      "rootSta ...
Error: Found '{' where a key name was expected (check your syntax or use quotes if the key name includes {}[],: or whitespace) at line 67,5 >>>    {
      "rootSta ...

```

</details>

**Comment by TheDutchCoder** [comment]
Our entire build is failing due to this very strange 6.4.0 -> 7.3.1 upgrade. It can't find certain parameters anymore.

Side note: this seems like an... unusual approach to semver?

**Comment by kaustavghosh06** [comment]
@kouz75 @TheDutchCoder Could it be possible for you'll to send your `amplify` folders to amplify-cli@amazon.com for us to investigate? We're closely monitoring it and will look into it right away.

**Comment by TheDutchCoder** [comment]
> @kouz75 @TheDutchCoder Could it be possible for you'll to send your `amplify` folders to [amplify-cli@amazon.com](mailto:amplify-cli@amazon.com) for us to investigate? We're closely monitoring it and will look into it right away.

I'd really rather not, but here's the full log of the failed build. Reverting to 6.4.0 makes it work again:
```
                                 # Starting phase: preBuild
                                 # Executing command: curl -X POST -H 'Content-type: application/json' --data '{"text":"['develop'] Amplify Backend deployment starting"}' [redacted]
2021-11-15T16:24:11.245Z [WARNING]: % Total    % Re
2021-11-15T16:24:11.245Z [WARNING]: ceived % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
                                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2021-11-15T16:24:11.296Z [WARNING]: 100    58    0     2  100    56
2021-11-15T16:24:11.297Z [WARNING]: 39   1098 --:--:-- --:--:-- --:--:--  1137
2021-11-15T16:24:11.297Z [INFO]: ok
2021-11-15T16:24:11.298Z [INFO]: # Completed phase: preBuild
                                 # Starting phase: build
2021-11-15T16:24:23.177Z [INFO]: [0mAmplify AppID found: [redacted]. Amplify App name is: [redacted][0m
2021-11-15T16:24:23.345Z [INFO]: [0mBackend environment staging found in Amplify Console app: [redacted][0m
2021-11-15T16:24:26.845Z [WARNING]: - Fetching updates to backend environment: staging from the cloud.
2021-11-15T16:24:29.798Z [WARNING]: - Building resource auth/userPoolGroups
2021-11-15T16:24:29.799Z [WARNING]: - Building resource auth/[redacted]
2021-11-15T16:24:29.802Z [WARNING]: - Building resource storage/s30625986b
2021-11-15T16:24:29.934Z [WARNING]: ✔ Successfully pulled backend environment staging from the cloud.
2021-11-15T16:24:30.608Z [INFO]: 
2021-11-15T16:24:35.654Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-11-15T16:24:41.663Z [WARNING]: - Initializing your environment: staging
2021-11-15T16:24:54.030Z [WARNING]: - Building resource auth/userPoolGroups
2021-11-15T16:24:54.034Z [WARNING]: - Building resource auth/[redacted]
                                    - Building resource storage/s30625986b
2021-11-15T16:24:54.035Z [WARNING]: ✔ Initialized provider successfully.
2021-11-15T16:25:53.021Z [WARNING]: - Uploading files...
2021-11-15T16:25:53.032Z [WARNING]: - Uploading files...
2021-11-15T16:25:53.314Z [WARNING]: - Uploading files...
2021-11-15T16:25:54.609Z [WARNING]: - Uploading files...
2021-11-15T16:25:54.614Z [WARNING]: - Uploading files...
2021-11-15T16:25:59.481Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.835Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.849Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.857Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.874Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.902Z [WARNING]: - Uploading files...
2021-11-15T16:26:00.930Z [WARNING]: - Uploading files...
2021-11-15T16:26:01.012Z [WARNING]: - Uploading files...
2021-11-15T16:26:04.402Z [WARNING]: - Uploading files...
2021-11-15T16:26:05.652Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.524Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.562Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.578Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.584Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.589Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.596Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.613Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.622Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.630Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.656Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.668Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.689Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.699Z [WARNING]: - Uploading files...
2021-11-15T16:26:07.893Z [WARNING]: - Uploading files...
2021-11-15T16:26:08.227Z [WARNING]: - Uploading files...
2021-11-15T16:26:08.568Z [WARNING]: - Uploading files...
2021-11-15T16:26:08.597Z [WARNING]: - Uploading files...
2021-11-15T16:26:12.303Z [INFO]: [32mGraphQL schema compiled successfully.[39m
                                 [32mEdit your schema at /codebuild/output/src709363363/src/[redacted]/amplify/backend/api/[redacted]/schema.graphql or place .graphql files in a directory at /codebuild/output/src709363363/src/[redacted]/amplify/backend/api/[redacted]/schema[39m
2021-11-15T16:26:13.116Z [WARNING]: - Uploading files...
2021-11-15T16:26:14.627Z [WARNING]: - Updating resources in the cloud. This may take a few minutes...
2021-11-15T16:26:20.997Z [INFO]: 
2021-11-15T16:26:20.999Z [INFO]: UPDATE_IN_PROGRESS amplify-[redacted]-staging-05753 AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:15 GMT+0000 (Coordinated Universal Time) User Initiated
2021-11-15T16:26:25.920Z [INFO]: 
2021-11-15T16:26:25.926Z [INFO]: UPDATE_IN_PROGRESS functionDefineAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPreSignup                   AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPostConfirmation            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPreTokenGeneration          AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionCreateAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionCreateAndInitBusiness       AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionSyncShopifyProducts         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS hostingamplifyhosting               AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionVerifyAuthChallengeResponse AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionDefineAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPostConfirmation            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPreTokenGeneration          AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionCreateAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPreSignup                   AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionVerifyAuthChallengeResponse AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionCreateAndInitBusiness       AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionSyncShopifyProducts         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    hostingamplifyhosting               AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:24 GMT+0000 (Coordinated Universal Time)
2021-11-15T16:26:30.954Z [INFO]: 
2021-11-15T16:26:30.956Z [INFO]: UPDATE_IN_PROGRESS [redacted] AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:28 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_FAILED      [redacted] AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:29 GMT+0000 (Coordinated Universal Time) Parameters: [functionDefineAuthChallengeName, functionPreTokenGenerationName, functionVerifyAuthChallengeResponseArn, functionPreSignupArn, functionPreSignupName, functionDefineAuthChallengeArn, functionPostConfirmationArn, functionVerifyAuthChallengeResponseName, functionCreateAuthChallengeArn, functionPreTokenGenerationArn, functionPostConfirmationName, functionCreateAuthChallengeName] do not exist in the template
2021-11-15T16:26:35.926Z [INFO]: 
2021-11-15T16:26:35.928Z [INFO]: UPDATE_ROLLBACK_IN_PROGRESS amplify-[redacted]-staging-05753 AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:31 GMT+0000 (Coordinated Universal Time) The following resource(s) failed to update: [[redacted]].
2021-11-15T16:26:50.953Z [INFO]: 
2021-11-15T16:26:50.955Z [INFO]: UPDATE_IN_PROGRESS functionCreateAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS hostingamplifyhosting               AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPreTokenGeneration          AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPostConfirmation            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionSyncShopifyProducts         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionVerifyAuthChallengeResponse AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionDefineAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionCreateAndInitBusiness       AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionCreateAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS functionPreSignup                   AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPreTokenGeneration          AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionSyncShopifyProducts         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionVerifyAuthChallengeResponse AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:48 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionDefineAuthChallenge         AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:49 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPostConfirmation            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:49 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionPreSignup                   AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:49 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    hostingamplifyhosting               AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:49 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    functionCreateAndInitBusiness       AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:49 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    [redacted]                   AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:50 GMT+0000 (Coordinated Universal Time)
2021-11-15T16:26:55.958Z [INFO]: 
2021-11-15T16:26:55.959Z [INFO]: UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS amplify-[redacted]-staging-05753 AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:54 GMT+0000 (Coordinated Universal Time)
2021-11-15T16:27:00.959Z [INFO]: 
2021-11-15T16:27:00.960Z [INFO]: UPDATE_COMPLETE          hostingamplifyhosting                  AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:56 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          [redacted]                      AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:56 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionCreateAndInitBusiness          AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:56 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionSyncShopifyProducts            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:56 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionCreateAuthChallenge            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionPreSignup                      AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionPostConfirmation               AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionPreTokenGeneration             AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionDefineAuthChallenge            AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE          functionVerifyAuthChallengeResponse    AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:57 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_ROLLBACK_COMPLETE amplify-[redacted]-staging-05753 AWS::CloudFormation::Stack Mon Nov 15 2021 16:26:58 GMT+0000 (Coordinated Universal Time)
2021-11-15T16:27:21.129Z [INFO]: Following resources failed
2021-11-15T16:27:21.132Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-11-15T16:27:21.151Z [WARNING]: ✖ There was an error initializing your environment.
2021-11-15T16:27:21.157Z [INFO]: [0mResourceNotReady: Resource is not in the state stackUpdateComplete[0m
                                 [0m    at constructor.setError (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/resource_waiter.js:182:47)[0m
                                 [0m    at Request.CHECK_ACCEPTORS (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/resource_waiter.js:44:12)[0m
                                 [0m    at Request.callListeners (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:106:20)[0m
                                 [0m    at Request.emit (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)[0m
                                 [0m    at Request.emit (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:686:14)[0m
                                 [0m    at Request.transition (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:22:10)[0m
                                 [0m    at AcceptorStateMachine.runTo (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:14:12)[0m
                                 [0m    at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:26:10[0m
                                 [0m    at Request.<anonymous> (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:38:9)[0m
                                 [0m    at Request.<anonymous> (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:688:12)[0m
                                 [0m    at Request.callListeners (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:116:18)[0m
                                 [0m    at Request.emit (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)[0m
                                 [0m    at Request.emit (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:686:14)[0m
                                 [0m    at Request.transition (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:22:10)[0m
                                 [0m    at AcceptorStateMachine.runTo (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:14:12)[0m
                                 [0m    at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:26:10[0m
                                 [0m    at Request.<anonymous> (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:38:9)[0m
                                 [0m    at Request.<anonymous> (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:688:12)[0m
                                 [0m    at Request.callListeners (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:116:18)[0m
                                 [0m    at callNextListener (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:96:12)[0m
                                 [0m    at IncomingMessage.onEnd (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/event_listeners.js:335:13)[0m
                                 [0m    at IncomingMessage.emit (events.js:326:22)[0m
                                 [0m    at IncomingMessage.EventEmitter.emit (domain.js:483:12)[0m
                                 [0m    at endReadableNT (_stream_readable.js:1241:12)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:84:21) {[0m
                                 [0m  code: 'ResourceNotReady',[0m
                                 [0m  retryable: false,[0m
                                 [0m  time: 2021-11-15T16:27:16.124Z,[0m
                                 [0m  statusCode: 200,[0m
                                 [0m  retryDelay: 30000[0m
                                 [0m}[0m
2021-11-15T16:27:21.211Z [ERROR]: !!! Build failed
2021-11-15T16:27:21.212Z [ERROR]: !!! Non-Zero Exit Code detected
2021-11-15T16:27:21.212Z [INFO]: # Starting environment caching...
2021-11-15T16:27:21.212Z [INFO]: # Uploading environment cache artifact...
2021-11-15T16:27:21.299Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by akshbhu** [comment]
hi @TheDutchCoder 

Can you share the cli-inputs.json file present in auth category folder after migration?
 Since this file contains your resource state , just redact any confidential info from it before sharing to``` amplify-cli@amazon.com```

**Comment by kaustavghosh06** [comment]
@TheDutchCoder Totally understand the confidentiality. 
If we could get the keys of your parameters.json object in amplify/backend/auth folder, that'll really help us reproduce the issue on our end and provide a fix.

**Comment by TheDutchCoder** [comment]
> @TheDutchCoder Totally understand the confidentiality. If we could get the keys of your parameters.json object in amplify/backend/auth folder, that'll really help us reproduce the issue on our end and provide a fix.

Sure thing, here you go:
```
{
    "identityPoolName": "[redacted]_identitypool",
    "allowUnauthenticatedIdentities": true,
    "resourceNameTruncated": "creati0626f80b",
    "userPoolName": "[redacted]_userpool",
    "autoVerifiedAttributes": [
        "email"
    ],
    "mfaConfiguration": "OFF",
    "mfaTypes": [
        "SMS Text Message"
    ],
    "smsAuthenticationMessage": "Your authentication code is {####}",
    "smsVerificationMessage": "Your verification code is {####}",
    "emailVerificationSubject": "Your verification code",
    "emailVerificationMessage": "Your verification code is {####}",
    "defaultPasswordPolicy": false,
    "passwordPolicyMinLength": 8,
    "passwordPolicyCharacters": [],
    "requiredAttributes": [
        "email",
        "family_name",
        "given_name"
    ],
    "userpoolClientGenerateSecret": true,
    "userpoolClientRefreshTokenValidity": 30,
    "userpoolClientWriteAttributes": [
        "email",
        "family_name",
        "given_name"
    ],
    "userpoolClientReadAttributes": [
        "email",
        "family_name",
        "given_name"
    ],
    "userpoolClientLambdaRole": "creati0626f80b_userpoolclient_lambda_role",
    "userpoolClientSetAttributes": true,
    "sharedId": "0626f80b",
    "resourceName": "[redacted]",
    "authSelections": "identityPoolAndUserPool",
    "authRoleArn": {
        "Fn::GetAtt": [
            "AuthRole",
            "Arn"
        ]
    },
    "unauthRoleArn": {
        "Fn::GetAtt": [
            "UnauthRole",
            "Arn"
        ]
    },
    "useDefault": "manual",
    "thirdPartyAuth": false,
    "usernameAttributes": [
        "email"
    ],
    "userPoolGroups": true,
    "adminQueries": true,
    "triggers": "{\n  \"CreateAuthChallenge\": [\n    \"custom\"\n  ],\n  \"DefineAuthChallenge\": [\n    \"custom\"\n  ],\n  \"PostConfirmation\": [\n    \"add-to-group\"\n  ],\n  \"PreSignup\": [\n    \"custom\"\n  ],\n  \"VerifyAuthChallengeResponse\": [\n    \"custom\"\n  ],\n  \"PreTokenGeneration\": [\n    \"alter-claims\"\n  ]\n}",
    "hostedUI": false,
    "userPoolGroupList": [
        "Users",
        "Admins"
    ],
    "adminQueryGroup": "Admins",
    "serviceName": "Cognito",
    "parentStack": {
        "Ref": "AWS::StackId"
    },
    "permissions": [
        "{\n  \"policyName\": \"AddToGroupCognito\",\n  \"trigger\": \"PostConfirmation\",\n  \"effect\": \"Allow\",\n  \"actions\": [\n    \"cognito-idp:AdminAddUserToGroup\",\n    \"cognito-idp:GetGroup\",\n    \"cognito-idp:CreateGroup\"\n  ],\n  \"resource\": {\n    \"paramType\": \"!GetAtt\",\n    \"keys\": [\n      \"UserPool\",\n      \"Arn\"\n    ]\n  }\n}"
    ],
    "dependsOn": [
        {
            "category": "function",
            "resourceName": "CreateAuthChallenge",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        },
        {
            "category": "function",
            "resourceName": "DefineAuthChallenge",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        },
        {
            "category": "function",
            "resourceName": "VerifyAuthChallengeResponse",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        },
        {
            "category": "function",
            "resourceName": "PreTokenGeneration",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        },
        {
            "category": "function",
            "resourceName": "PreSignup",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        },
        {
            "category": "function",
            "resourceName": "PostConfirmation",
            "triggerProvider": "Cognito",
            "attributes": [
                "Arn",
                "Name"
            ]
        }
    ]
}
```

**Comment by akshbhu** [comment]
Hi @TheDutchCoder 

Can you also check  amplify/cli.json file if  it has breakcirculardependency flag  set to true?

**Comment by TheDutchCoder** [comment]
> Hi @TheDutchCoder
> 
> Can you also check amplify/cli.json file if it has breakcirculardependency flag set to true?

It doesn't. Here's the config:
```json
{
  "features": {
    "graphQLTransformer": {
      "addMissingOwnerFields": true,
      "validateTypeNameReservedWords": true,
      "enableIterativeGSIUpdates": true,
      "secondaryKeyAsGSI": true
    },
    "auth": {
      "enableCaseInsensitivity": true,
      "useInclusiveTerminology": true
    },
    "appSync": {
      "generateGraphQLPermissions": true
    },
    "codegen": {
      "useAppSyncModelgenPlugin": true,
      "useDocsGeneratorPlugin": true,
      "useTypesGeneratorPlugin": true,
      "cleanGeneratedModelsDirectory": true,
      "retainCaseStyle": true
    }
  }
}
```

**Comment by kaustavghosh06** [comment]
@TheDutchCoder Could you give it a try in Amplify Console with latest CLI version and confirm if the issue has been mitigated?

**Comment by TheDutchCoder** [comment]
> @TheDutchCoder Could you give it a try in Amplify Console with latest CLI version and confirm if the issue has been mitigated?

We'll give it a shot tomorrow in the AM and I'll report back!

**Comment by TheDutchCoder** [comment]
> @TheDutchCoder Could you give it a try in Amplify Console with latest CLI version and confirm if the issue has been mitigated?

Yes, the issue seems to have been fixed by 7.4.0. Thanks guys!

What's the exact reason for this, somewhat unorthodox, semver approach btw?

**Comment by kaustavghosh06** [comment]
@TheDutchCoder Glad that you are  unblocked!

We did a major version bump since we did an internal change to support overrides of resources and custom resources for the Amplify CLI (https://aws.amazon.com/amplify/extensibility/) which would directly impact customer files (within the amplify folder) in some cases. Whenever we do customer impacting changes which might potentially require file-changes (and we try to auto-migrate all these changes), we do a major version bump for the CLI.
